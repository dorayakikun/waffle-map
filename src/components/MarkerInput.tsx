import * as React from "react";
import { Button, Input, InputOnChangeData, Message } from "semantic-ui-react";

export interface Props {
  latLng: string;
  errorMessage: string;
  putMarker: (latLng: string) => void;
  removeAllMarkers: () => void;
}

export interface State {
  latLng: string;
  errorMessage: string;
}

export function MarkerInput(props: Props) {
  const [latLng, setLatLng] = React.useState<string>(props.latLng || "");

  const handleChange = React.useCallback(
    (_event: React.SyntheticEvent<HTMLElement>, data: InputOnChangeData) =>
      setLatLng(data.value),
    []
  );

  const handleClickPutAMarker = React.useCallback(
    (event: any) => {
      if (
        (event as React.KeyboardEvent<HTMLElement>).key === "Enter" ||
        (event as React.MouseEvent<HTMLElement>).type === "click"
      ) {
        props.putMarker(latLng);
      }
    },
    [latLng]
  );

  const handleClickRemoveAllMarkers = React.useCallback(
    () => props.removeAllMarkers(),
    []
  );

  return (
    <div onKeyPress={handleClickPutAMarker}>
      <Input
        error={props.errorMessage !== ""}
        inverted
        onChange={handleChange}
        placeholder="lat,lng"
        style={{
          marginBottom: "10px",
          marginRight: "3px",
          marginTop: "10px"
        }}
        value={latLng}
      />

      <Button icon="marker" onClick={handleClickPutAMarker} />
      <Button icon="trash" onClick={handleClickRemoveAllMarkers} />

      {props.errorMessage !== "" && (
        <Message negative>
          <Message.Header>Waffle Map Error</Message.Header>
          <p>{props.errorMessage}</p>
        </Message>
      )}
    </div>
  );
}
