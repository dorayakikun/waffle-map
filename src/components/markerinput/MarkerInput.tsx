import { AddIcon, DeleteIcon } from "@chakra-ui/icons";
import {
  Button,
  FormControl,
  FormErrorMessage,
  HStack,
  Input,
  Stack,
} from "@chakra-ui/react";
import * as React from "react";

export type Props = {
  errorMessage: string;
  latLngString: string;
  onPutMarkerClicked: (e: any) => void;
  onRemoveAllMarkersClicked: () => void;
};

export function MarkerInput(props: Props) {
  const [latLngString, setLatLngString] = React.useState<string>(
    props.latLngString || ""
  );

  const handleChange = React.useCallback(
    (e: React.SyntheticEvent<HTMLElement>) =>
      setLatLngString((e.target as any).value),
    []
  );

  return (
    <FormControl
      isInvalid={props.errorMessage !== ""}
      onKeyPress={props.onPutMarkerClicked}
    >
      <HStack spacing={3}>
        <Input
          onChange={handleChange}
          placeholder="lat,lng"
          value={latLngString}
        />
        <Stack direction="row" spacing={3}>
          <Button
            leftIcon={<AddIcon />}
            colorScheme={"teal"}
            onClick={props.onPutMarkerClicked}
            width="128px"
            variant="outline"
          >
            Put
          </Button>
          <Button
            leftIcon={<DeleteIcon />}
            colorScheme={"teal"}
            onClick={props.onRemoveAllMarkersClicked}
            width="128px"
            variant="outline"
          >
            Remove
          </Button>
        </Stack>
      </HStack>
      <FormErrorMessage>{props.errorMessage}</FormErrorMessage>
    </FormControl>
  );
}
