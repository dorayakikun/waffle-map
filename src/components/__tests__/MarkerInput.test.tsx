import * as React from "react";
import * as render from "react-test-renderer";
import { MarkerInput, Props } from "../MarkerInput";

const props: Props = {
  errorMessage: "",
  latLng: "",
  putMarker: () => {},
  removeAllMarkers: () => {}
};

test("Should set props to MarkerInput", () => {
  const tree = render.create(<MarkerInput {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should set props to MarkerInput(datum is Tokyo)", () => {
  const errorMessage = "It seems there was something wrong ...";
  const latLng = "lat,lng";
  const newProps = { ...props, errorMessage, latLng };
  const tree = render.create(<MarkerInput {...newProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
