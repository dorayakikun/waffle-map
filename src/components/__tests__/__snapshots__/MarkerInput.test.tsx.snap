// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should set props to MarkerInput 1`] = `
<div
  onKeyPress={[Function]}
>
  <div
    className="ui inverted input"
    style={
      Object {
        "marginBottom": "10px",
        "marginRight": "3px",
        "marginTop": "10px",
      }
    }
  >
    <input
      onChange={[Function]}
      placeholder="lat,lng"
      type="text"
      value=""
    />
  </div>
  <button
    className="ui icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="marker icon"
      onClick={[Function]}
    />
  </button>
  <button
    className="ui icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="trash icon"
      onClick={[Function]}
    />
  </button>
</div>
`;

exports[`Should set props to MarkerInput(datum is Tokyo) 1`] = `
<div
  onKeyPress={[Function]}
>
  <div
    className="ui error inverted input"
    style={
      Object {
        "marginBottom": "10px",
        "marginRight": "3px",
        "marginTop": "10px",
      }
    }
  >
    <input
      onChange={[Function]}
      placeholder="lat,lng"
      type="text"
      value="lat,lng"
    />
  </div>
  <button
    className="ui icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="marker icon"
      onClick={[Function]}
    />
  </button>
  <button
    className="ui icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="trash icon"
      onClick={[Function]}
    />
  </button>
  <div
    className="ui negative message"
  >
    <div
      className="header"
    >
      Waffle Map Error
    </div>
    <p>
      It seems there was something wrong ...
    </p>
  </div>
</div>
`;
